UOP1_Project/Assets/Scripts/StateMachine/Initializable.cs

```diff
@@ -0,0 +1,17 @@
+﻿using UnityEditor;
+using UnityEngine;
+
+namespace KarimCastagnini.PluggableFSM
+{
+	//A class to perform initialization only when in PlayMode or when in Built, need to find a more elegant way
+	public abstract class Initializable : ScriptableObject
```

Comment:
One alternative could be to use an interface and an extension method:

```csharp
public interface Initializable
{
    void Initialize();
}

public static class InitializableExtensions
{
    public static void HandleInitialization(this Initializable initializable)
    {
        if (EditorApplication.isPlayingOrWillChangePlaymode || Application.isPlaying)
            initializable.Initialize();
    }
}
```

And then on each class that needs initialization implement Initializable interface:

```csharp
public class StateBase : ScriptableObject, Initializable
{
    public void Initialize()
    {
        // Do stuff...
    }

    private void OnEnable()
    {
        HandleInitialization();
    }
}
```